### PR TITLE
feat(extensions): add cli commands

### PR DESCRIPTION
**Pull request by syu-w**
Add new commands expand-extensions and list-extensions

Fix: https://github.com/canonical/charmcraft/issues/1242

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -704,7 +705,7 @@ def test_systemdetails_hidden_auth():
 
 # -- generic tests for all Charmcraft commands
 
-all_commands = list.__add__(*[cgroup.commands for cgroup in COMMAND_GROUPS])
+all_commands = list(itertools.chain(*(cgroup.commands for cgroup in COMMAND_GROUPS)))
```
Comment (lengau) [inline tests/test_main.py:708]: Did this come from a linter issue? If not, please remove this change here (though a separate PR with this fix would be much appreciated!)

Comment (syu-w) [reply tests/test_main.py:708]: It is a invalid code when `COMMAND_GROUPS` has more than 2 items.
`TypeError: expected 1 argument, got 2`

File: charmcraft/commands/extensions.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+_overview = """
+Extensions allow user use pre-defined configuration in their charmcraft.yaml.
+"""
+
+
+class ListExtensionsCommand(BaseCommand):
+    """List available extensions for all supported bases."""
+
+    common = True
+    name = "list-extensions"
```
Comment (lengau) [inline charmcraft/commands/extensions.py:40]: Please include a hidden `extensions` version of this command (see the plurals section at https://discourse.ubuntu.com/t/command-structure/18556).

**Review by lengau — CHANGES_REQUESTED** [review]

File: tests/test_main.py (hunk 1 above)
Comment (lengau) [reply tests/test_main.py:708]: Ahh, I see. All good then, thanks!

**Review by sergiusens — APPROVED** [review]

**Pull request merged**